### PR TITLE
Add integration test for querying from disk and fix bug.

### PR DESCRIPTION
**Pull request by notbdu**
Adds an integration test for querying raw data from disk.

Fixes a bug where reading results from disk resulted in duplicate results. #146

### CODE REVIEW

File: values/decoding/raw_size_string_iterator.go [hunk 1]
```diff
@@ -53,7 +52,7 @@ func (it *rawSizeStringIterator) Next() bool {
 		return false
 	}
 
-	it.curr = unsafe.ToString(it.buf[:rawSizeBytes])
+	it.curr = string(it.buf[:rawSizeBytes])
```
Comment (xichen2020) [inline values/decoding/raw_size_string_iterator.go]: `safe.ToString`

File: x/hash/hash.go [hunk 2]
```diff
@@ -22,7 +22,7 @@ func BytesHash(d []byte) Hash {
 
 // StringHash returns the hash of a string.
 func StringHash(s string) Hash {
-	return BytesHash(unsafe.ToBytes(s))
+	return BytesHash(safe.ToBytes(s))
```
Comment (xichen2020) [inline x/hash/hash.go]: Given the usage here is confined with a single line within a single function, and the converted bytes are not used outside the function at all, it should be fine to use `unsafe` in this simple context.

**Comment by xichen2020** [comment]
Seems ci is failing due to permission error, you probably should use the temp directory as the root directory for db data, but not sure if that's directly related

**Comment by notbdu** [comment]
That's weird... wonder how it was creating the data directory before...

**Comment by notbdu** [comment]
Oh jk, it wasn't flushing data to disk in the other integration tests before haha.

**Pull request merged**